### ISSUE TITLE
about weight L1_loss compute?

### ISSUE THREAD

**Issue by Johnson-yue** [zhyxun/Separating-Style-and-Content-for-Generalized-Style-Transfer#2]
Hi, thank you for your share,  In your two paper << Separating Style and Content for Generalized Style Transfer  >> and << A unified Framework for Generalizable Style Transfer: Style and Content Separation >>

about **W_st** as you paper,  it is defined as the reciprocal of the number of black pixels in each target image.   And **N_b** is the number of black pixel . I think your dataset  image is black character with white backgroud,  so the **black pixel** should smaller than 0.5 (you normal data range is [0,1])  but in your [code](https://github.com/zhyxun/Separating-Style-and-Content-for-Generalized-Style-Transfer/blob/f99717a07c74e063e86cb194b952923cc6db4127/train.py#L17) it is bigger than 0.5 ? Why?

**Comment by zhyxun** [comment]
@Johnson-yue  Thank you for your comment. Yes, the values of black pixels should be smaller than 0.5.  In the code, the tf.greater(0.5, batch_targets_holder) is used.  tf.greater(a,b) is true when a>b.  So N_b is calculated as the number of pixels whose values are smaller than 0.5.

**Comment by Johnson-yue** [comment]
Hi, I use your weighted_l1 loss compare with standard l1 loss, they are very similar performance. 
when thin font like(line 2):
![image](https://user-images.githubusercontent.com/10268274/53282949-ed1a3900-3779-11e9-979f-7e020b70680e.png)
line 1 : is content image which including 10 images with the same content and different style.
line 2 : is style image which including 10 images with the same style but different content.
line 3: is target
line 4: is generated by model

Do you have any idea with it ??

**Comment by zhyxun** [comment]
What is the dataset you trained on? 

----- 原始邮件 -----
发件人: "Johnson yue" <notifications@github.com>
收件人: "zhyxun/Separating-Style-and-Content-for-Generalized-Style-Transfer" <Separating-Style-and-Content-for-Generalized-Style-Transfer@noreply.github.com>
抄送: "zhyxun" <zhyxun@sjtu.edu.cn>, "Comment" <comment@noreply.github.com>
发送时间: 星期六, 2019年 2 月 23日 下午 2:50:00
主题: Re: [zhyxun/Separating-Style-and-Content-for-Generalized-Style-Transfer] about weight L1_loss compute? (#2)

Hi, I use your weighted_l1 loss compare with standard l1 loss, they are very similar performance. 
when thin font like(line 2):
![image](https://user-images.githubusercontent.com/10268274/53282949-ed1a3900-3779-11e9-979f-7e020b70680e.png)
line 1 : is content image which including 10 images with the same content and different style.
line 2 : is style image which including 10 images with the same style but different content.
line 3: is target
line 4: is generated by model

Do you have any idea with it ??


-- 
You are receiving this because you commented.
Reply to this email directly or view it on GitHub:
https://github.com/zhyxun/Separating-Style-and-Content-for-Generalized-Style-Transfer/issues/2#issuecomment-466623106
-- 
Best Regards,
Yexun
=======================================
Shanghai Jiao Tong University
Cooperative Medianet Innovation Center
Room 303A, No.5 SEIEE Building
Shanghai 200240, China

Mobile: +86 176 9226 3268
Email:  zhyxun@sjtu.edu.cn
=======================================

**Comment by zhyxun** [comment]
What is the dataset you trained on?

**Comment by Johnson-yue** [comment]
> What is the dataset you trained on?

the dataset is my own hand-write dataset

**Comment by zhyxun** [comment]
It may be because the dataset you trained on is not large enough. What's the size of the dataset and how many kinds of styles/contents are there in the dataset? Besides, during our experiment, it is also harder to generate hand-writting characters with only a few reference images.

**Comment by Johnson-yue** [comment]
my dataset including 124 hand-write font and which font have 6763 characters.
I want to use only 825 or less character generator others.

But, my question is: Is the weighted  L1 loss useful ?
I compared weighted-L1 and standard-L1, and the result is the same,  Maybe I have wrong way with weighted-L1, is it ??

**Comment by Alijin** [comment]
@Johnson-yue Hi, I have seen the same paper and project, but now I have a problem on how to run it. I run it on python 3.5. Could you mind giving me some advice please?

**Comment by Alijin** [comment]
@zhyxun @Johnson-yue Hi, I have seen the same paper and project, but now I have a problem on how to run it. I run it on python 3.5. Could you mind giving me some advice please?

**Comment by zhyxun** [comment]
@Alijin I run it on Python2.7 and TensorFlow 1.4.0.

**Comment by Alijin** [comment]
@zhyxun OK，thank you very much

**Comment by Alijin** [comment]
@Johnson-yue could you mind sharing the project that you run well?

**Comment by suker80** [comment]
L1 loss don't decrease can you help me?

i used 430 fonts  and 1300 characters

('model_loss', 1.7846603)
('mse', 0.6581955)
('l1 loss', 2265.267)

model loss converage  1.7 

how can you load your dataset?

**Comment by zhyxun** [comment]
@suker80 All pixels of training images are transfromed into [0,1]. During our experiment, the "model_loss" will decrease from >1 to about 0.2~0.3. Maybe there is something wrong with your implementation.

**Comment by suker80** [comment]
@zhyxun How long does it take to complete the training?

**Comment by zhyxun** [comment]
@suker80 Using the dataset described in the paper, we need ~30h using one GPU to complete the training.

**Comment by suker80** [comment]
@zhyxun thank you for your help ! code works very well

generated images looks fine !

**Comment by ersanliqiao** [comment]
> @suker80 Using the dataset described in the paper, we need ~30h using one GPU to complete the training.

想问下，数据集能共享一下吗？另外，能描述下，几个真值文件的组织形式吗？

**Comment by ersanliqiao** [comment]
@zhyxun 想问下，数据集能共享一下吗？另外，能描述下，几个真值文件的组织形式吗？

**Comment by ersanliqiao** [comment]
> @suker80 Using the dataset described in the paper, we need ~30h using one GPU to complete the training.

How much data did you need 30h?

**Comment by ishvindersethi22** [comment]
@suker80 can you please share the repository and dataset, It would be a great help!

**Comment by ersanliqiao** [comment]
> @zhyxun thank you for your help ! code works very well
> 
> generated images looks fine !

How is your result of novel styles

**Comment by otori-bird** [comment]
@Johnson-yue @suker80 
could you tell me how to organize the data?
i have my own 3000 hand-written images but i dont know how to train it

**Comment by Johnson-yue** [comment]
only 3000 images？？？ it absolutely not enough for trainin

**Comment by otori-bird** [comment]
@Johnson-yue i have totally 34000 or more images from different authors. The author's style that i'd like to train has about 3000

**Comment by ishvindersethi22** [comment]
@Johanson-yue can you share the dataset you've used?

**Comment by Johnson-yue** [comment]
@ishvindersethi22  Sorry I can not share my dataset
